### DISCUSSION TITLE
Why FlexSearch don't change over to TypeScript?

### DISCUSSION THREAD

**Discussion by ts-thomas** [nextapps-de/flexsearch#419] [General]
There is just one reason why I don't see any advantage to use TypeScript. The Closure Compiler simply produces better Javascript, compiles the code more aggressively. It is not the size (nobody cares about size when using gzip), it is the performance gain the Closure Compiler is able to archive.

If anyone can show me a build setup for TypeScript which produces better optimized Javascript, then I will go for it for sure.

Assume we have a full scoped typescript definition index.d.ts, do you have any disadvantages? How we could improve it without changing over to Typescript?

**Comment by Sondro** [comment]
The main benefit of TypeScript are the data types. When used in Electron-esque apps they perform better or may crash without types specified (because the compiler guesses wrong or undefined, which means more manual work later). TS UI frameworks like React are used more & more in desktop & mobile/ARM based apps like games.

**Reply by ooker777** [reply]
As I'm a newbie I don't really understand it much. But isn't that if you don't want TS, you just need to delete the d.ts files?
